gtsam/discrete/DecisionTree.h

```diff
@@ -113,14 +119,19 @@ namespace gtsam {
     convert(const typename DecisionTree<M, X>::NodePtr& f, const std::map<M,
         L>& map, std::function<Y(const X&)> op);
 
-    /** Default constructor */
-    DecisionTree();
+    /** Convert only node to a different type */
+    template <typename X>
+    NodePtr convert(const typename DecisionTree<L, X>::NodePtr& f,
```

Comment:
Where are these convert things used? I frown at the non-functional style, even though I probably did this myself. Do we really need it? If not, we can deprecate them…

Reply:
I don't think they are used anywhere other than in the constructor for converting to another type. We can deprecate them but we will still need a mechanism to convert the nodes from one type to another (for the `linearize` method atleast).

Reply:
Linearize is functional, though.  I don't understand why we are adding a non-const method. You say constructor but I don't see a new constructor...

Reply:
It's in the constructor definitions in `DecisionTree-inl.h`.

Reply:
I see. As long as the converts are private, all is good. If they are not, can you move the declarations to the private block at the end?

Reply:
Yup, they are in `protected` so we should be good. :)